lightning/src/ln/features.rs

```diff
@@ -251,28 +265,34 @@ mod sealed {
 					const ASSERT_ODD_BIT_PARITY: usize = (<Self as $feature>::ODD_BIT % 2) - 1;
 				}
 			)*
+
 		}
 	}
 
 	define_feature!(1, DataLossProtect, [InitContext, NodeContext],
-		"Feature flags for `option_data_loss_protect`.");
+		"Feature flags for `option_data_loss_protect`.", set_supports_data_loss_protect,
+		set_requires_data_loss_protect);
```

Comment:
As per my earlier comment, I'd prefer `set_required_` and `set_optional_` prefixes. Or maybe `set_<feature>_required` and `set_<feature>_optional` would read better?